pipeline/src/pipeline.rs

```diff
@@ -808,7 +808,7 @@ impl<T: FieldElement> Pipeline<T> {
         let query_callback = self
             .arguments
             .query_callback
-            .take()
```

Comment:
Otherwise the query callback is only available for the first-phase witness generation.